pkgs/development/libraries/sqlcipher/default.nix

```
@@ -13,14 +13,15 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-E23PTNnVZbBQtHL0YjUwHNVUA76XS8rlARBOVvX6zZw=";
   };
 
-  nativeBuildInputs = [ installShellFiles ];
-
-  buildInputs = [ readline ncurses openssl tcl ];
+  nativeBuildInputs = [ installShellFiles tcl ];
+  buildInputs = [ readline ncurses openssl ];
+  depsBuildBuild = [ buildPackages.stdenv.cc ];
 
   configureFlags = [ "--enable-threadsafe" "--disable-tcl" ];
 
   CFLAGS = [ "-DSQLITE_ENABLE_COLUMN_METADATA=1" "-DSQLITE_SECURE_DELETE=1" "-DSQLITE_ENABLE_UNLOCK_NOTIFY=1" "-DSQLITE_HAS_CODEC" ];
   LDFLAGS = lib.optional (readline != null) "-lncurses";
+  BUILD_CC = "$(CC_FOR_BUILD)";
```

I think the following is a bit easier to follow, at least not as weird as having make-style variable expansions in ordinary environment variables:

```suggestion
  makeFlags = [ "BUILD_CC=$(CC_FOR_BUILD)" ];
```